pyxrf/xanes_maps/xanes_maps_api.py

```diff
@@ -1002,27 +1072,71 @@ def adjust_incident_beam_energies(scan_energies, emission_line):
 #   Functions used for processing
 
 def _get_uniform_grid(positions_x_all, positions_y_all):
-    """Compute uniform grid common to the whole dataset"""
-    # Median positions are probably the best for generating common uniform grid
+    """
+    Compute common uniform grid based on position coordinates for a stack of maps.
+    The grid is computed in two steps: the median of X and Y coordinates is computed
+    for each point of the map; uniform grid is generated based on median X and Y values.
```

Comment:
I have trouble understanding how the median is being used. Is each point of the map a different energy?

Reply:
We don't do any operations that involve energy. The issue here is that the positioning system is not perfect and there are random fluctuations of positions, so the idea is to get some grid, which represents position of the sample in the best way. Also I believe that sometimes the sequence of scans is interrupted for alignment. Therefore positions may shift along X and Y axes between consecutive scans. The estimated positions are not considered reliable and used in the program only for plotting output maps.

Reply:
I changed the comment a little bit, so I guess it is easier to understand.